core/modules/query/BoolTerm.cc

```diff
@@ -219,12 +219,13 @@ bool BoolFactor::_checkParen(BfTerm::PtrList& terms) {
 
 boost::shared_ptr<BoolTerm> BoolFactor::getReduced() {
     // Get reduced versions of my children.
-    BfTerm::PtrList newTerms;
+    BfTerm::PtrVector newTerms;
     bool hasReduction = false;
     hasReduction = _reduceTerms(newTerms, _terms);
     // Parentheses reduction
     if(_checkParen(newTerms)) {
-        newTerms.pop_front();
+        assert(!newTerms.empty());
```

Comment:
I do not think that assert will do anything useful here, `_checkParen` already guarantees what you need.